combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/NodeBleDevice.kt

```diff
@@ -0,0 +1,117 @@
+package inc.combustion.framework.ble.device
+
+import android.bluetooth.BluetoothAdapter
+import android.util.Log
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import inc.combustion.framework.LOG_TAG
+import inc.combustion.framework.ble.scanning.CombustionAdvertisingData
+import inc.combustion.framework.ble.uart.meatnet.*
+import inc.combustion.framework.ble.uart.meatnet.NodeReadFeatureFlagsRequest
+import inc.combustion.framework.service.DebugSettings
+import inc.combustion.framework.service.DeviceConnectionState
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+internal class NodeBleDevice(
+    mac: String,
+    owner: LifecycleOwner,
+    private var nodeAdvertisingData: CombustionAdvertisingData,
+    adapter: BluetoothAdapter,
+    private var uart: UartBleDevice = UartBleDevice(mac, nodeAdvertisingData, owner, adapter)
+) {
+
+    protected val genericRequestHandler = UartBleDevice.MessageCompletionHandler()
+    protected val readFeatureFlagsRequest = UartBleDevice.MessageCompletionHandler()
+
+    companion object {
+        const val NODE_MESSAGE_RESPONSE_TIMEOUT_MS = 5000L
+    }
+
+    init {
+        processUartResponses()
+    }
+
+    val rssi: Int get() { return uart.rssi }
```

Comment:
No worries if this is a pain in the ass, this is just a suggestion.  

It might be nice to be able to force this interface in case of changes to it in the future.  I think the inheritance would be something like  `NodeBleDevice <- CombustionBleDevice` and `ProbeBleDevice <- CombustionBleDevice`.  

And pulling the common interface from `ProbeBleDevice` (for instance, [here](https://github.com/combustion-inc/combustion-android-ble/blob/develop/combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/ProbeBleDeviceBase.kt#L68-L73)), into `CombustionBleDevice` so that it is maintained in one spot.

Reply:
I'll look into this a little bit to see how much work it would be.

Reply:
I like this idea. I'm going to split it out into a separate ticket for some follow-on work.